### PR TITLE
pycosat 0.6.4 synced from conda-forge

### PR DESCRIPTION
**Pull request by dholth**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Update version; sync from conda-forge; fix recipe branch name

### CODE REVIEW

**Comment by dholth** [comment]
The first version of this referenced the wrong tag. It looks like there are some additional issues because the branch was renamed to main.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+  sha256: 79ffa99156810e58876dd6374af0b7bff88f9a4580ca8b8468c414c5c2454bb8
 
 build:
-  number: 1
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
 
 requirements:
   build:
+    - python                                 # [build_platform != target_platform]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:18]: We do not provide cross compiling so these two lines can be removed

Comment (skupr-anaconda) [inline recipe/meta.yaml:18]: We do not provide cross compiling so these two lines can be removed

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,22 +30,30 @@ test:
...
   imports:
     - pycosat
+  commands:
+    - pip check
+    - python test_package.py "{{ version }}"
 
 about:
-  home: https://github.com/ContinuumIO/pycosat
+  home: https://github.com/conda/pycosat
   license: MIT
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:47]: Missing `license_family: MIT`

File: recipe/meta.yaml [hunk 3]
```diff
@@ -28,22 +30,30 @@ test:
...
+  license_file:
+    - LICENSE
+    - COPYRIGHT
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Only LICENSE file is available https://github.com/conda/pycosat/blob/0.6.4/LICENSE

**Comment by skupr-anaconda** [comment]
@dholth `conda_build.exceptions.CondaBuildException: Found a build.sh script and a build/script section inside meta.yaml. Either remove the build.sh script or remove the build/script section in meta.yaml.`

File: recipe/meta.yaml (hunk 1 above)
Comment (dholth) [reply recipe/meta.yaml:18]: would it be OK to stay more equal to the conda-forge recipe

Comment (skupr-anaconda) [reply recipe/meta.yaml:18]: It can be but we removed them for our feedstocks

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**